### ISSUE TITLE
Add how-to setup .edgerc / EdgeGrid Authentication in readme.md

### ISSUE THREAD

**Issue by akamaimike** [akamai/akamai-docker#6]
Can you please add how-to instructions in the readme.md that expains how a user can add their Akamai EdgeGrid to the docker container? I did this by creating the .edgerc file with vi in the root. 

Example documentation to add: 

### How to add EdgeGrid Authentication
To ensure, requests and actions are properly authenticated, please create a .edgerc file in the root containing your unique combination of host, client token, client secret and access token. More information on creating an API Client can be found here: [https://developer.akamai.com/api/getting-started](Getting Started with Akamai APIs)

`vi .edgerc`

Add in your EdgeGrid Authentication, labeling with the correct section in brackets. 

`[default]`
`client_secret = 'your_client_secret'`
`host = 'your_host'`
`access_token = 'your_access_token'`
`client_token = 'your_client_token'`

To verify, if authentication is working properly, please run the Akamai CLI request:

`akamai auth verify`

**Comment by lukaszczerpak** [comment]
I am not sure creating `.edgerc` inside the container is a good practice.
If you use the image for experimenting, then it's more convenient to map local `.edgerc` as follows:

```
docker run --rm -it -v $HOME/.edgerc:/root/.edgerc akamai/akamai-docker akamai auth verify
```

In CI/CD pipeline credentials are typically exposed via environment variables and it can be a separate step to create such configuration file in the workspace, especially if there are more commands to be invoked from the image.

@ynohat thoughts?

**Comment by ynohat** [comment]
How the credentials are handled will largely depend on the use case, but I feel that @lukaszczerpak is correct; common practice is to treat containers as mostly ephemeral, the edgerc file should therefore be mounted directly from the host using the `-v ~/.edgerc:/root/.edgerc:ro` option, which avoids those manual steps every time the container is re-created, and also improves the security of the setup.

I believe there is value in being explicit about this in the README, however.

@akamaimike what do you think?

**Comment by akamaimike** [comment]
This has already been eye-opening, perhaps it is my lack of Docker experience. I agree with treating the Docker container as an ephemeral/non-changing item and mounting on the authentication. 

The main use-case from my perspective is getting customer to use this Docker container as an everything-in-one Akamai toolbox for starting their journey, making basic requests and then later on start on creating complex actions.  

This can be during an Akamai University/Developer World Tour session, at a conference/event or during workshops to get them started with the APIs (through HTTPie), Akamai CLI, Akamai Terraform etc.

**Comment by ynohat** [comment]
Yes, we're looking at two main use cases:

* all-in-one interactive shell with dependencies pre-installed for training and exploration
* small, specialized images, one per module, for automation work (cf. #2), where size and dependency control matter

Let's keep this open and fix the README to explicitly state our approach to managing the edgerc.

**Comment by ynohat** [comment]
@akamaimike, @lukaszczerpak could you please chime in/fix on the PR #7 ?

**Comment by akamaidev** [comment]
Closing as this has been implemented and merged
